stempy/blockiterator.cpp

```
@@ -0,0 +1,66 @@
+
+#include "blockiterator.h"
+
+#include <iostream>
+
+using std::cout;
+using std::cerr;
+using std::endl;
+
+using std::string;
+using std::vector;
+
+namespace stempy {
+
+BlockIterator::BlockIterator(const vector<string>& fileList)
+{
+  m_streams.reserve(fileList.size());
+  for (const auto& file: fileList)
+    m_streams.push_back(StreamReader(file));
+
+  if (m_streams.empty())
+    cerr << "Warning: no files provided for BlockIterator" << endl;
```

For this sort of error we should throw an exception.